### PR TITLE
refactor: put ips into separate module

### PR DESCRIPTION
**Pull request by asmie**
IPS has been separated from the rest of crunchy.
This move has two major benefits:
- separated code is more readable and don't mix with other crunchy functionalities;
- allow to grab IPS as a separate product more easily.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -16,7 +12,7 @@ use ziggurat_core_crawler::summary::NetworkSummary;
 use crate::{
     config::CrunchyConfiguration,
     geoip_cache::GeoIPCache,
-    ips::Ips,
+    ips::ips_algorithm::Ips,
```
Comment (Rqnsom) [inline src/main.rs:15]: one less keyword, wdyt?
```suggestion
    ips::algorithm::Ips,
```

**Review by Rqnsom — APPROVED** [review]
LGTM

File: src/main.rs (hunk 1 above)
Comment (asmie) [reply src/main.rs]: Done.

**Review by kylegranger — APPROVED** [review]
👍

**Pull request merged**